server/market/types/types.go

```diff
@@ -0,0 +1,151 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/decred/dcrdex/server/asset"
+)
+
+// MarketInfo specified a market that the Archiver must support.
+type MarketInfo struct {
+	Name    string
+	Base    uint32
+	Quote   uint32
+	LotSize uint64
+}
+
+func marketName(base, quote string) string {
+	return base + "_" + quote
+}
+
+func MarketName(base, quote uint32) (string, error) {
+	baseSymbol := asset.BipIDSymbol(base)
+	if baseSymbol == "" {
+		return "", fmt.Errorf("base asset %d not found", base)
+	}
+	baseSymbol = strings.ToLower(baseSymbol)
+
+	quoteSymbol := asset.BipIDSymbol(quote)
+	if quoteSymbol == "" {
+		return "", fmt.Errorf("quote asset %d not found", quote)
+	}
+	quoteSymbol = strings.ToLower(quoteSymbol)
+
+	return marketName(baseSymbol, quoteSymbol), nil
+}
+
+func NewMarketInfo(base, quote uint32, lotSize uint64) (*MarketInfo, error) {
+	name, err := MarketName(base, quote)
+	if err != nil {
+		return nil, err
+	}
+	return &MarketInfo{
+		Name:    name,
+		Base:    base,
+		Quote:   quote,
+		LotSize: lotSize,
+	}, nil
+}
+
+func NewMarketInfoFromSymbols(base, quote string, lotSize uint64) (*MarketInfo, error) {
+	baseID, found := asset.BipSymbolID(strings.ToLower(base))
+	if !found {
+		return nil, fmt.Errorf(`base asset symbol "%s" unrecognized`, base)
+	}
+
+	quoteID, found := asset.BipSymbolID(strings.ToLower(quote))
+	if !found {
+		return nil, fmt.Errorf(`quote asset symbol "%s" unrecognized`, quote)
+	}
+
+	return &MarketInfo{
+		Name:    marketName(base, quote),
+		Base:    baseID,
+		Quote:   quoteID,
+		LotSize: lotSize,
+	}, nil
+}
+
+// OrderStatus indicates how an order is presently being processed, or the final
+// state of the order if processing is completed.
+type OrderStatus uint16
+
+const (
+	// OrderStatusUnknown is a sentinel value to be used when the status of an
+	// order cannot be determined.
+	OrderStatusUnknown OrderStatus = iota
+
+	// There are two general classes of orders: ACTIVE and TERMINAL. Orders with
```

Comment:
I should change "terminal" to "archived" since I use "archived" everywhere else.